### PR TITLE
NSString/ComparisonTests.PrefixSuffix/8 fails on OSX

### PR DESCRIPTION
**Pull request by bbowman**
[ RUN      ] NSString/ComparisonTests.PrefixSuffix/8
Foundation/ReferenceFoundation/TestNSString.mm:910: Failure
Value of: [lhs hasPrefix:rhs]
  Actual: '\0'
Expected: expectHasPrefix
Which is: '\x1' (1)
[  FAILED  ] NSString/ComparisonTests.PrefixSuffix/8, where GetParam() = (, , '\x1' (1), '\0', '\x1' (1), '\x1' (1), '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0') (0 ms)

### CODE REVIEW

File: tests/unittests/Foundation/ReferenceFoundation/TestNSString.mm [hunk 1]
```diff
@@ -977,13 +980,17 @@ virtual void TearDown() {
     combinerWithLhs.attach([combiner mutableCopy]);
     [combinerWithLhs appendString:lhs];
 
+    LOG_INFO("lhs - \"%s\"\n rhs - \"%s\"\n", [lhsWithFragment description], [rhs description]);
     checkHasPrefixHasSuffix(lhsWithFragment, rhs, ::testing::get<6>(GetParam()), ::testing::get<7>(GetParam()));
+    LOG_INFO("lhs - \"%s\"\n rhs - \"%s\"\n", [fragmentWithLhs description], [rhs description]);
     checkHasPrefixHasSuffix(fragmentWithLhs, rhs, ::testing::get<8>(GetParam()), ::testing::get<9>(GetParam()));
+    LOG_INFO("lhs - \"%s\"\n rhs - \"%s\"\n", [lhsWithCombiner description], [rhs description]);
     checkHasPrefixHasSuffix(lhsWithCombiner, rhs, ::testing::get<10>(GetParam()), ::testing::get<11>(GetParam()));
+    LOG_INFO("lhs - \"%s\"\n rhs - \"%s\"\n", [combinerWithLhs description], [rhs description]);
```
Comment (bbowman) [inline tests/unittests/Foundation/ReferenceFoundation/TestNSString.mm]: Probably should remove these.

**Comment by bbowman** [comment]
@DHowett-MSFT is added to the review. #Closed

**Comment by bbowman** [comment]
@rajsesh-msft is added to the review. #Closed

**Comment by bbowman** [comment]
@ms-jihua is added to the review. #Closed

**Comment by msft-Jeyaram** [comment]
:shipit:

**Pull request merged**